azext_iot/iothub/providers/message_endpoint.py

```diff
@@ -247,13 +253,66 @@ def list(self, endpoint_type: Optional[str] = None):
         elif EndpointType.AzureStorageContainer.value == endpoint_type:
             return endpoints.storage_containers
 
-    def delete(self, endpoint_name: Optional[str] = None, endpoint_type: Optional[str] = None):
+    def delete(
+        self,
+        endpoint_name: Optional[str] = None,
+        endpoint_type: Optional[str] = None,
+        force: bool = False
+    ):
         endpoints = self.hub_resource.properties.routing.endpoints
         if endpoint_type:
             endpoint_type = endpoint_type.lower()
             if EndpointType.CosmosDBContainer.value == endpoint_type and not self.support_cosmos:
                 raise InvalidArgumentValueError(INVALID_CLI_CORE_FOR_COSMOS)
 
+        if self.hub_resource.properties.routing.enrichments or self.hub_resource.properties.routing.routes:
+            # collect endpoints to remove
+            endpoint_names = []
+            if endpoint_name:
+                endpoint_names.append(endpoint_name)
```

Comment:
What happens when the only input endpoint_name is provided for a cosmos db container and self.support_cosmos is False?

Reply:
If self.support_cosmos is false, the cosmos db endpoints will not be checked and thus, the routes and message enrichments for those will not be checked.

However, I do see the problem with endpoint_name (and I am honestly not sure what the service is doing in this scenario). I will make it so that there is an extra check here